subservers/subserver.go

```diff
@@ -0,0 +1,152 @@
+package subservers
```

Comment:
nit: is there a big difference in the contents of this commit (other than adding things to a separate package) along with the next 3 commits compared to the original PR's commits? wondering If I  could get a cheeky commit credit or two 🤓